### ISSUE TITLE
Appium test error with protractor and saucelabs

### ISSUE THREAD

**Issue by DenisBiondic** [zalando/zalenium#563]
Zalenium Image Version(s):  
dosel/zalenium:**3.11.0g** & elgalu/selenium:**3.11.0-p11**

Docker Version:  
18.03.1-ce

OS: 
Windows 10

Docker Command to start Zalenium:  
`docker run --rm -ti --name zalenium -p 4444:4444 -e SAUCE_USERNAME=xxx -e SAUCE_ACCESS_KEY=xxx -v /var/run/docker.sock:/var/run/docker.sock -v /tmp/videos:/home/seluser/videos --privileged dosel/zalenium start --sauceLabsEnabled true --startTunnel true`

## Expected Behavior -
Test should run without timeouts when using following capability with saucelabs:

```
    {
      "appiumVersion": "1.7.1",
      "deviceName": "iPad 2 Simulator",
      "deviceOrientation": "portrait",
      "platformVersion": "9.3",
      "platformName": "iOS",
      'browserName': 'safari'
    },
```

## Actual Behavior -
Test times out, error message from sauce: "Your test errored. Test did not see a new command for 90 seconds. Timing out."

Few more infos:
- as a test client I am using protractor with a very basic angular app
- protractor.conf is

```
const { SpecReporter } = require('jasmine-spec-reporter');

exports.config = {
  allScriptsTimeout: 11000,
  seleniumAddress: 'http://localhost:4444/wd/hub',

  specs: [
    './e2e/**/*.e2e-spec.ts'
  ],

  multiCapabilities : [
    {
      "appiumVersion": "1.7.1",
      "deviceName": "iPad 2 Simulator",
      "deviceOrientation": "portrait",
      "platformVersion": "9.3",
      "platformName": "iOS",
      'browserName': 'safari'
    },
  ],

  baseUrl: 'http://localhost:4200/',
  framework: 'jasmine',
  jasmineNodeOpts: {
    showColors: true,
    defaultTimeoutInterval: 30000,
    print: function() {}
  },
  onPrepare() {
    require('ts-node').register({
      project: 'e2e/tsconfig.e2e.json'
    });
    jasmine.getEnv().addReporter(new SpecReporter({ spec: { displayStacktrace: true } }));
  }
};
```

- test works fine when using connection to saucelabs directly, via specifying the sauce credentials directly in the protractor. Test errors only when going through the zalenium / selenium grid / sauce.
- zalenium works fine when specifying capabilities such as chrom / firefox / internet explorer. IE for example is forwarded to sauce labs without errors.

**Comment by pearj** [comment]
Can you post the zalenium logs?

**Comment by DenisBiondic** [comment]
> See the complete license at https://github.com/zalando/zalenium/blob/master/LICENSE.md
*************************************** Data Processing Agreement ***************************************
08:56:10.335 [Thread-12] INFO  d.z.e.z.registry.ZaleniumRegistry - Registered a node http://172.17.0.3:40000
08:56:10.353 [Thread-12] INFO  d.z.e.z.proxy.AutoStartProxySet - Registered a container 3d57dc8a83366062dec74518c3bd663e34206d50663ac557da08d56fe2b1bc6c http://172.17.0.3:40000.
08:56:10.483 [Thread-13] INFO  d.z.e.z.registry.ZaleniumRegistry - Registered a node http://172.17.0.4:40000
08:56:10.484 [Thread-13] INFO  d.z.e.z.proxy.AutoStartProxySet - Registered a container 44aedc08aa48fbecf9e0415aa83d7711a2dac13382281b158ec41bb5cf541d77 http://172.17.0.4:40000.
08:57:20.910 [qtp52514534-12] INFO  o.o.g.w.s.handler.RequestHandler - Got a request to create a new session: Capabilities {appiumVersion: 1.7.1, browserName: safari, count: 1, deviceName: iPad 2 Simulator, deviceOrientation: portrait, platformName: ios, platformVersion: 9.3}
08:57:20.913 [qtp52514534-12] INFO  d.z.e.z.registry.ZaleniumRegistry - Adding sessionRequest for {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator}
08:57:20.915 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=9c09c5d8-73f4-4648-a34a-8358f5493ae2, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=chrome, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=66.0.3359.139, platform=LINUX}
08:57:20.916 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=f84ed260-97f0-4cfe-80cd-36d5ca366179, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=firefox, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=59.0.3, platform=LINUX}
08:57:20.917 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=6dceff36-d09e-4ed4-8901-90e70fa3c424, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=chrome, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=66.0.3359.139, platform=LINUX}
08:57:20.918 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=1c8456b0-3eed-4d7e-94cc-7d4aa674ebc8, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=firefox, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=59.0.3, platform=LINUX}
08:57:20.928 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=9c09c5d8-73f4-4648-a34a-8358f5493ae2, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=chrome, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=66.0.3359.139, platform=LINUX}
08:57:20.929 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=f84ed260-97f0-4cfe-80cd-36d5ca366179, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=firefox, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=59.0.3, platform=LINUX}
08:57:20.930 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=6dceff36-d09e-4ed4-8901-90e70fa3c424, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=chrome, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=66.0.3359.139, platform=LINUX}
08:57:20.930 [Matcher thread] INFO  d.z.e.z.m.DockerSeleniumCapabilityMatcher - Validating {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator} in node with capabilities {server:CONFIG_UUID=1c8456b0-3eed-4d7e-94cc-7d4aa674ebc8, seleniumProtocol=WebDriver, acceptSslCerts=true, screen-resolution=1920x1080, tz=Europe/Berlin, browserName=firefox, maxInstances=1, platformName=LINUX, screenResolution=1920x1080, resolution=1920x1080, version=59.0.3, platform=LINUX}
08:57:20.932 [Matcher thread] INFO  d.z.e.z.p.CloudTestingRemoteProxy - Test will be forwarded to SauceLabs, {appiumVersion=1.7.1, platformVersion=9.3, count=1, browserName=safari, deviceOrientation=portrait, platformName=ios, deviceName=iPad 2 Simulator}
08:57:20.936 [Matcher thread] INFO  org.openqa.grid.internal.TestSlot - Trying to create a new session on test slot {browserName=SauceLabs, maxInstances=2, platformName=ANY}
08:57:20.938 [Matcher thread] INFO  d.z.e.z.registry.ZaleniumRegistry - Test session with internal key cd80513a-22ca-4d6b-90c7-522467e77b31 assigned to remote () after 0 seconds (139 ms).
08:57:20.955 [Forwarding cd80513a-22ca-4d6b-90c7-522467e77b31 (int. key, remote not contacted yet.) to http://ondemand.saucelabs.com:80/wd/hub at 8:57:20 AM] INFO  d.z.e.z.p.CloudTestingRemoteProxy - Currently using 1 parallel sessions towards SauceLabs. Attempt to start one more.
08:58:51.279 [Thread-11] INFO  o.o.grid.internal.ActiveTestSessions - Removed a session that had not yet assigned an external key cd80513a-22ca-4d6b-90c7-522467e77b31, indicates failure in session creation ORPHAN
08:58:51.279 [Thread-11] INFO  d.z.e.z.p.CloudTestingRemoteProxy - SauceLabs Releasing slot and terminating session due to inactivity.

**Comment by DenisBiondic** [comment]
saucelabs commands:

![image](https://user-images.githubusercontent.com/3341262/39982915-2d744f72-5756-11e8-8e00-8655006f0105.png)

**Comment by DenisBiondic** [comment]
I also tried the debug log but it outputs too much and I didn't find a way to direct to output to a log file instead of the console. Probably have to check if possible through logback.xml. Do you need the debug logs as well?

**Comment by DenisBiondic** [comment]
Do you need a sample project for easier reporoduction? I can quickly create an angular / protractor project here on github with all the commands to reproduce.

**Comment by diemol** [comment]
@DenisBiondic yes, that'd be great since I haven't had time to build one on my own. Thanks!

**Comment by diemol** [comment]
Hi again, 

I tried locally with a Java project and a Protractor one and it works fine.

The difference with the posted config is that I didn't have my system under test on localhost, which is what I think the given configuration has: `baseUrl: 'http://localhost:4200/`.

There is an issue there, where localhost cannot be used with the tunnel to test through Zalenium, this in the case the operating system is not Linux. So, it does not work in OSX and Windows. Back then when we checked it was a docker issue but we would be happy to get more insights of it.

Better use something different than localhost, if possible.

I will close this issue since, so far, a solution would be out of the scope of Zalenium.
